en/05_Contributing/05_Coding_Conventions/01_PHP_Coding_Conventions.md

```diff
@@ -136,6 +136,133 @@ class MyClass extends Class
+- Avoid the use of the splat `...` parameter as this makes API enforcement difficult. If you need to support multiple arguments then use an array parameter type.
+
+## Object instantiation
+
+- Use Injector to instantiate objects rather than the `new` keyword. This is done as it allows projects to swap out classes with different implementations.
+- Note there is only a minimal performance penalty for this, though there is a downside that IDE type hinting stops working. During development you may wish to use the new keyword for this reason and change it to use injector before submitting your PR.
+- For classes that use the Injectable trait use the `::create()` static method.
+- For objects that do not have the Injectable trait, use `Injector::inst()->get({classname})`
+- This also includes instantiation of third party classes `<<< @todo why? discuss`
+- The exception to this is for unit-tests where you should use the `new` because there you need to ensure you're testing a specific class
+
+## Extensions and traits
+
+- Use the DataExtension class for extending DataObjects.
+- Use the Extension class for extending non-DataObjects.
+- Typically you shouldn't use native PHP traits. A common gotcha is mistakenly using a trait on a DataObject to add a private static config property. This doesn't work because private static config properties are a Silverstripe specific thing and the Silverstripe DataExtension will correctly merge them. However traits are unable to merge private static config properites.
+
+### Extension hooks
+
+- Naming convention for extension hooks is `update{methodName}` where methodName has any leading `get` omitted has the remaining first letter uppercased.
+
+### Unit testing
+
+- The unit test method name should match the method that it's testing with the first letter of the method uppercased and the word `test` prefixed e.g. for the method `myMethodName()` the unit test method should be called `testMyMethodName()`
+- Aim to use DataProvider's to provide test case data as it makes code much cleaner and it makes it very easy to add additional test cases.
+- The DataProvider method name should be same as the test case method name it's providing for with the leading work `test` substituted for `provide` e.g. for `testSomething()` the DataProvider is `provideSomething()`
+- Typically you should use yml fixture files for creating fixtures, though this is a fairly loose recommendation. Often it will make sense to "create them manually" in PHP.
+- Using reflection to change the visibility of a private or protected method so that it can be independently tested is usually OK. At a practical level sometimes only testing public methods is overly clunky.
+
+### Admin-only controllers
+
+- Controllers that are only intended to be used from within the CMS should extend [`LeftAndMain`](api:SilverStripe\Admin\LeftAndMain). This will handle routing for you and ensures that only authenticated users are able to access their endpoints.
```

Comment:
```suggestion
- Controllers that are only intended to be used from within the CMS should extend [`LeftAndMain`](api:SilverStripe\Admin\LeftAndMain). This will handle routing for you and it ensures that only authenticated users are able to access their endpoints.
```

Reply:
I don't agree with this. `LeftAndMain` is way overused for that purpose and the fact you have to hide the section in `_config.php` and don't use probably a large majority of `LeftAndMain` functionality in your given controller is a massive code smell. I think we should leave this out as a "standard". We don't have to create new admin controller often anyway, and next time we do create one we should look at finding a cleaner way to do it.

And frankly the chances of anyone not on the CMS Squad wanting to do this in a PR is highly unlikely so there's no harm in not documenting this until we have a better way to do it, IMO.

Reply:
Have removed the section